File: frontend/models/order.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Order struct {
	ID              primitive.ObjectID `bson:"_id,omitempty"`
	ProductName     string             `bson:"productName" validate:"required"`
	ProductImage    string             `bson:"productImage"`
	Stock           int32              `bson:"stock" validate:"required"`
	ClientName      string             `bson:"clientName" validate:"required"`
	ClientEmail     string             `bson:"clientEmail" validate:"required,email"`
	Address         string             `bson:"address" validate:"required"`
	Status          string             `bson:"status" validate:"required,oneof=Pending Delivered"`
	QuantityOrdered int32              `bson:"quantityOrdered" validate:"required"`
	CreatedAt       primitive.DateTime `bson:"createdAt,omitempty"`
}

File: backend/models/Order.js
const mongoose = require('mongoose');

const OrderSchema = new mongoose.Schema(
    {
        productName: {
            type: String, 
            required: true
        },
        productImage: {
            type: String,
            required: true
        },
        stock: {
            type: Number,
            required: true
        },
        clientName: {
            type: String, 
            required: true
        },
        clientEmail: {
            type: String,
            required: true
        },
        address: {
            type: String,
            required: true
        },
        status: {
            type: String,
            default: 'Pending',
            enum: ['Pending', 'Delivered'] 
        },
        quantityOrdered: {
            type: Number,
            required: true
        }
    },
    {
        timestamps: true
    },
);

module.exports = mongoose.model('Order', OrderSchema);
// En los argumentos del model:
//nombre del modelo en sigular 'Orders' y luego el squema que lo representa 